File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/analysis/acceptance-project/src/surveycrawler/counter_clockwise_turn.py
from surveycrawler.crawler_turn import CrawlerTurn


class CounterClockwiseTurn(CrawlerTurn):

    # turning

    def apply_to(self, a_crawler):
        a_crawler.facing().turn_counter_clockwise(a_crawler)

    def inverted(self):
        from surveycrawler.clockwise_turn import ClockwiseTurn

        return ClockwiseTurn()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-214358-38088/analysis/own-tests-project/src/main/java/surveycrawler/CounterClockwiseTurn.java
package surveycrawler;

public class CounterClockwiseTurn extends CrawlerTurn {

    // applying

    @Override
    public void applyTo(SurveyCrawler aCrawler) {
        aCrawler.turnFacingCounterClockwise();
    }

    // undoing

    @Override
    public CrawlerAction opposite() {
        return CrawlerTurn.clockwise();
    }
}
